--- a/partdiff.c	2021-10-30 16:31:12.600140999 +0200
+++ b/partdiff.c	2021-10-30 16:30:07.047627922 +0200
@@ -344,9 +344,14 @@
 			/* over all columns */
 			for (j = 1; j < N; j++)
 			{
-				star = (Matrix[m2][i - 1][j] + Matrix[m2][i][j - 1] + Matrix[m2][i][j + 1] + Matrix[m2][i + 1][j]) / 4;
+				star = Matrix[m2][i - 1][j] + Matrix[m2][i][j - 1] + Matrix[m2][i][j + 1] + Matrix[m2][i + 1][j];
+				if (options->inf_func == FUNC_FPISIN)
+			        {
+					double const h = arguments->h;
+			                star += (2 * M_PI * M_PI) * h * h * sin(M_PI * h * (double)i) * sin(M_PI * h * (double)j);
+			        }
 
-				star += calculate_func(arguments, options, i, j);
+				star /= 4;
 
 				residuum    = Matrix[m2][i][j] - star;
 				residuum    = fabs(residuum);
